### PR TITLE
SEQ-7207: parallel assembly.

### PR DESCRIPTION
**Pull request by sbandara**
This PR is for parallelization of assembly, alignment, and genotyping across many nodes. To this end, I extracted that execution path from `run_metasv` into `asm_sc_intervals`, which is called by `run_metasv` when in standalone mode. To allow for different run modes, `--disable_assembly` is replaced (albeit still supported) by `--assembly=?`, where `?` is either `run`, `disable`, `defer`, `parallel`, or `merge`. The last three values summon the new features, as briefly explained in the changes to `index.html`. Those new run modes come with additional arguments for slicing work across nodes and for which sliced result files to merge. In addition to `run_metasv`, `main.py` now also has `run_distributed_assembly` and `run_merge_assembly_slices` to support the new modes of execution. These entry points are resolved by the existing launcher function in `scripts/run_metasv`. The should not affect existing workflows and features. For example, the alternative entry points in `run_spades.py` and `run_age.py` still exist, and `--nthreads` can still be used, e.g. people who use LSF with `-n` can use multiple threads on each parallel worker. Please do not merge yet - intended for review, discussion, testing, and fixes.

### CODE REVIEW

**Comment by marghoob** [comment]
Thanks @sbandara. The parallelization changes are similar to what I had in mind, especially the case checks for nthreads > 1.

File: metasv/run_spades.py [hunk 1]
```diff
@@ -170,11 +165,16 @@ def run_spades_parallel(bam=None, spades=None, spades_options="", bed=None, work
         return None, None
 
     bedtool = pybedtools.BedTool(bed)
-    total = bedtool.count()
 
     chrs = set(chrs)
     all_intervals = [interval for interval in bedtool] if not chrs else [interval for interval in bedtool if
                                                                          interval.chrom in chrs]
+    if slicing is not None:
```
Comment (marghoob) [inline metasv/run_spades.py]: Since None evaluates to false and you also want to skip slicing = [], you can simply have `if slicing:`

**Comment by marghoob** [comment]
Looks good to me for now.

**Comment by msahraeian** [comment]
Looks good to me too.

**Comment by sbandara** [comment]
Thanks everyone for the quick review! Before merging, we should still check off the following:
- [ ] Resolve RES-119. It looks like any fix will also take care of RES-113. Merge conflicts unlikely here because changes probably localized to `generate_sv_intervals.py` only.
- [ ] Much more testing with different slicing parameters on both Mac and Linux. Will certainly do that as part of SEQ-7163.

**Comment by marghoob** [comment]
Any timeline on this, @sbandara ? I am planning on upping the MetaSV version tomorrow.

**Comment by sbandara** [comment]
Sure, I can back-merge and resolve the conflicts tonight.

**Comment by sbandara** [comment]
Limping behing on this one too much. It would be nice if I could contribute in smaller chunks soon.

**Pull request closed without merging**